connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java

```diff
@@ -86,18 +86,40 @@ public ConnectorsResource(Herder herder) {
     @POST
     @Path("/")
     public Response createConnector(final @QueryParam("forward") Boolean forward,
-                                    final CreateConnectorRequest createRequest) throws Throwable {
-        String name = createRequest.name();
-        if (name.contains("/")) {
-            throw new BadRequestException("connector name should not contain '/'");
-        }
+                                     final CreateConnectorRequest createRequest) throws Throwable {
         Map<String, String> configs = createRequest.config();
-        if (!configs.containsKey(ConnectorConfig.NAME_CONFIG))
-            configs.put(ConnectorConfig.NAME_CONFIG, name);
+        CreateConnectorRequest finalCreateRequest = createRequest;
+
+        String nameFromBody = createRequest.name();
+        String nameFromConfig = createRequest.config().get(ConnectorConfig.NAME_CONFIG);
+        String name = "";
+
+        if (nameFromBody == null && nameFromConfig == null) {
```

Comment:
When working on this I rediscovered what made me write these conditions, which was these two lines from [ConnectorsResource.java](https://github.com/apache/kafka/blob/e31c0c9bdbad432bc21b583bd3c084f05323f642/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java#L95):

`if (!configs.containsKey(ConnectorConfig.NAME_CONFIG))
            configs.put(ConnectorConfig.NAME_CONFIG, name);`

These make me think that the name parameter at the top level of the request and the name parameter in the config sub-element are the same thing.
With this code, it is possible to create something like the following:

```
 {
	"name": "testconn2",
	"config": {
		"name": "badname",
		"connector.class": "org.apache.kafka.connect.tools.MockSourceConnector",
		"tasks.max": "1",
		"topics": "test-topic"
	},
	"tasks": []
}
```
Which may or may not be what is wanted..if a connector requires a name config and that needs to be different from the connector name this may be useful, but it is also prone to misunderstandings I think.

I am tempted to leave a check in place that disallows passing in two different name elements.
Any thoughts?